portal-bridge/src/bridge/history.rs

```diff
@@ -149,21 +149,22 @@ impl HistoryBridge {
             BridgeMode::Single(val) => (false, val),
             _ => panic!("Invalid backfill mode"),
         };
-        let (mut start_block, mut end_block, mut epoch_index) = match mode_type {
-            // end block will be same for an epoch in single & backfill modes
-            ModeType::Epoch(epoch_number) => (
-                epoch_number * EPOCH_SIZE,
-                ((epoch_number + 1) * EPOCH_SIZE),
-                epoch_number,
-            ),
+        let (start_block, mut end_block) = match mode_type {
+            ModeType::Epoch(epoch_number) => {
+                let end_block = match looped {
+                    true => latest_block,
+                    false => (epoch_number + 1) * EPOCH_SIZE,
+                };
+                (epoch_number * EPOCH_SIZE, end_block)
+            }
             ModeType::Block(block) => {
-                let epoch_index = block / EPOCH_SIZE;
                 let end_block = match looped {
```

Comment:
We're not using `looped` anymore. At least in the context of looping over gossip ranges. We should get rid of the variable, or rename it if it's still useful for setting other variable values